quantum/process_keycode/process_auto_shift.c

```diff
@@ -247,26 +207,28 @@ bool process_auto_shift(uint16_t keycode, keyrecord_t *record) {
             case KC_ASOFF:
                 autoshift_disable();
                 return true;
+        }
+    }
 
+    switch (keycode) {
 #    ifndef NO_AUTO_SHIFT_ALPHA
-            case KC_A ... KC_Z:
+        case KC_A ... KC_Z:
 #    endif
 #    ifndef NO_AUTO_SHIFT_NUMERIC
-            case KC_1 ... KC_0:
+        case KC_1 ... KC_0:
 #    endif
 #    ifndef NO_AUTO_SHIFT_SPECIAL
-            case KC_TAB:
-            case KC_MINUS ... KC_SLASH:
-            case KC_NONUS_BSLASH:
+        case KC_TAB:
+        case KC_MINUS ... KC_SLASH:
+        case KC_NONUS_BSLASH:
 #    endif
+            if (record->event.pressed) {
                 return autoshift_press(keycode, record);
-
-            default:
-                break;
-        }
+            } else {
+                autoshift_end(keycode, record->event.time, false);
+                return false;
```

Comment:
What would it take to rewrite this so that we could return true here (at least for the non-custom case)?

I tried to avoid eating records since the behaviour becomes less transparent to things downstream (like `process_record_user`).

Reply:
Basically just not mergng this. It's near impossible to return true and still have control of keyrepeat to make this not a breaking change, and this refactor relies heavily on the assumption that it will be the one pressing and releasing keys (as I was preparing for custom ones).

IMO that is more of a QMK-wide issue. We need a way to let other processes know that a key was pressed but that it's been handled and they shouldn't press anything, avoiding `return false;` everywhere is a pain.